spec/models/trading_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe Trading, type: :model do
+  describe 'validations' do
+    it 'validates presence of amount' do
+      trading = Trading.new
+      expect(trading).not_to be_valid
+    end
+
+    it 'validates numericality of amount' do
+      trading = Trading.new(amount: 'abc')
+      expect(trading).not_to be_valid
+    end
+  end
```

Comment:
ditto

Reply:
wrote validations and invalidation test for all vaiya